docs/configuration.md

```diff
@@ -79,88 +81,108 @@ spec:
-- `address`: The local address to bind API on. Also used as one of the addresses pushed on the k0s create service certificate on the API. Defaults to first non-local address found on the node.
-- `sans`: List of additional addresses to push to API servers serving certificate
-- `extraArgs`: Map of key-values (strings) for any extra arguments you wish to pass down to Kubernetes api-server process
-- `port`: custom port for kube-api server to listen on (default value 6443)
-- `k0sApiPort`: custom port for k0s-api server to listen on (default value 9443)
+## `spec` Key Detail
 
-Keep in mind, in case if `port` and `k0sApiPort` are used with `externalAddress` setting, the LB serving at `externalAddress` must listen on the same ports.  
-
-### `spec.controllerManager`
+### `spec.api`
 
-- `extraArgs`: Map of key-values (strings) for any extra arguments you wish to pass down to Kubernetes controller manager process
+| Element   | Description           |
+|-----------|---------------------------|
+| `externalAddress`      | The loadbalancer address (for k0s controllers running behind a loadbalancer). Configures all cluster components to connect to this address and also configures this address for use  when joining new nodes to the cluster.|   
+| `address`      | Local address on wihich to bind an API. Also serves as one of the addresses pushed on the k0s create service certificate on the API. Defaults to first non-local address found on the node.| 
+| `sans`      | List of additional addresses to push to API servers serving the certificate.| 
+| `extraArgs`      | Map of key-values (strings) for any extra arguments to pass down to Kubernetes api-server process.| 
+| `port`<sup>*</sup>     | Custom port for kube-api server to listen on (default: 6443)| 
+| `k0sApiPort`<sup>*</sup>     | Custom port for k0s-api server to listen on (default: 9443)| 
 
-### `spec.scheduler`
 
-- `extraArgs`: Map of key-values (strings) for any extra arguments you wish to pass down to Kubernetes scheduler process
+<sup>*</sup> If `port` and `k0sApiPort` are used with the `externalAddress` element, the loadbalancer serving at `externalAddress` must listen on the same ports.  
 
 ### `spec.storage`
 
-- `type`: Type of the data store, either `etcd` or `kine`.
-- `etcd.peerAddress`: Nodes address to be used for etcd cluster peering.
-- `kine.dataSource`: [kine](https://github.com/rancher/kine/) datasource URL.
-
-Using type `etcd` will make k0s to create and manage an elastic etcd cluster within the controller nodes.
+| Element   | Description           |
+|-----------|---------------------------|
+| `type`      | Type of the data store (valid values:`etcd` or `kine`).<br><br>**Note**: Type `etcd` will cause k0s to create and manage an elastic etcd cluster within the controller nodes.|   
+| `etcd.peerAddress`      | Nodes address used for etcd cluster peering.|   
```

Comment:
```suggestion
| `etcd.peerAddress`      | Node's address used for etcd cluster peering.|   
```

Reply:
Actually, it is better as: "Node adress used for etcd clusger peering."